scrape/scrape_test.go

```diff
@@ -3148,6 +3157,11 @@ func TestTargetScraperScrapeOK(t *testing.T) {
 			timeout := r.Header.Get("X-Prometheus-Scrape-Timeout-Seconds")
 			require.Equal(t, expectedTimeout, timeout, "Expected scrape timeout header.")
 
+			traceparent := r.Header.Get("traceparent")
+			require.NotEmpty(t, traceparent, "Expected traceparent header to be present")
+			require.Regexp(t, traceparentPattern, traceparent,
```

Comment:
let's use existing otel utils for such checks https://github.com/open-telemetry/opentelemetry-go/blob/bf7ffa3900bdda6253fa7653d478a75a3781343a/trace/trace.go#L87

nit: no need to add error messages to `require.xxx` here, the provided context is enough